### PR TITLE
Undeprecate securityadmin script

### PR DESCRIPTION
**Pull request by derek-ho**
### Description
There was some recent engagement/concern from the community (for example @spapadop) about the deprecation of security admin script. Since there is no concrete plan to remove/replace it, we should remove the deprecation warning. We can add this back in if/when we decide to and have a viable alternative for the script. 

### Issues Resolved
None
Is this a backport? If so, please add backport PR # and/or commits #, and remove `backport-failed` label from the original PR.
No

Do these changes introduce new permission(s) to be displayed in the static dropdown on the front-end? If so, please open a draft PR in the security dashboards plugin and link the draft PR here
No

### Testing
Script no longer prints the warning out
### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [ ] New Roles/Permissions have a corresponding security dashboards plugin PR
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md)
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/security/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by derek-ho** [comment]
@peternied I saw that in your original issue [you mentioned that this tool would continue to be supported through out 3.0.0](https://github.com/opensearch-project/security/issues/1755#issue-1197834994). We currently don't have bandwidth to come up with an alternative, so wdyt about removing this deprecation notice until if/when we have a path forward to deprecation, to avoid community panic?

**Comment by cwperks** [comment]
@derek-ho wdyt about reverting all of https://github.com/opensearch-project/security/pull/1756?

**Comment by derek-ho** [comment]
> @derek-ho wdyt about reverting all of #1756?

Sure I can add those as well. Initially I only added securityadmin as this is the main script that folks had concerns about.

**Review by peternied — COMMENTED** [review]
This warning was put into place as these tools aren't really supported - they aren't up to the quality bar we have for the rest of the security codebase - unless something has changed that I've missed?

I'll leave this to other maintainers to approve or not, but I don't think anything has changed that means these tools _can_ be depended on by any downstream project.

**Comment by cwperks** [comment]
Now that the `install_demo_configuration` has moved from completely being a [bash script to becoming a bash script that invokes a java program](https://github.com/opensearch-project/security/pull/3669), it has become unit-testable and up to the same quality as the rest of the codebase. 

> they aren't up to the quality bar we have for the rest of the security codebase - unless something has changed that I've missed?

@peternied can you expand on this?

Other tools in the tools folder have been receiving bug fixes and new features as well. They are actively maintained. Here's an example of a recent update to the Hasher tool: https://github.com/opensearch-project/security/pull/4524

**Review by cwperks — APPROVED** [review]

**Review by DarshitChanpura — APPROVED** [review]

**Comment by DarshitChanpura** [comment]
Let's un-deprecate these until there is a clear deprecation path and replacement in-place.

**Pull request merged**
